presto-main/src/main/java/com/facebook/presto/operator/project/PageProcessor.java

```diff
@@ -103,6 +105,17 @@ public PageProcessorOutput process(ConnectorSession session, Page page)
         return projections;
     }
 
+    private static long calculatePageRetainedSizeWithoutLoading(Page page)
+    {
+        long retainedSizeInBytes = 0;
+        for (Block block : page.getBlocks()) {
+            if (!(block instanceof LazyBlock)) {
+                retainedSizeInBytes += block.getRetainedSizeInBytes();
```

Comment:
👍  but this deserves some explanatory comment in the code. In the unlikely case that the next person here won't be you :)